source/Cosmos.HAL2/BlockDevice/EBR.cs

```diff
@@ -40,8 +41,8 @@ protected void ParsePartition(byte[] aEBR, UInt32 aLoc)
             }
             else if (xSystemID != 0)
             {
-                UInt32 xStartSector = aEBR.ToUInt32(aLoc + 8);
-                UInt32 xSectorCount = aEBR.ToUInt32(aLoc + 12);
+                UInt32 xStartSector = (uint)BitConverter.ToInt32(aEBR, (int)aLoc + 8);
```

Comment:
It should be `ToUInt32` here.

Reply:
ToUInt32 is not working. I am getting an Arg_InvalidOperationException when trying to use it. I think it has to be plugged.